### PR TITLE
38 parse the plane primitive

### PR DESCRIPTION
**Pull request by AxelF44**
I've added the parsing of the plane !

### CODE REVIEW

**Review by mathieurobert1 — DISMISSED** [review]
Seems good :+1:

File: include/Parser/Factory.hpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 #include "Parser/DLLoader.hpp"
 #include "Primitives/Sphere.hpp"
+#include "Primitives/Plane.hpp"
```
Comment (Njord201) [inline include/Parser/Factory.hpp:20]: Why include Plane.hpp in this file ?

File: src/Core/Parser/Scene.cpp [hunk 2]
```diff
@@ -132,6 +132,44 @@ int Raytracer::Scene::_parsePrimitiveSetting(const libconfig::Setting &primitive
...
     }
+
+    if (primitives.exists("planes")) {
+        libconfig::Setting& planeArray = primitives.lookup("planes");
+        for (int index = 0; index < planeArray.getLength(); index++) {
+            std::shared_ptr<Primitive::IPrimitive> plane = _factory.createPrimitivesComponent("plane");
+            std::shared_ptr<Primitive::Plane> newPlane = std::dynamic_pointer_cast<Primitive::Plane>(plane);
+
+            Math::Point3D position(0, 0, 0);
+
```
Comment (Njord201) [inline src/Core/Parser/Scene.cpp:143]: Remove one empty line

File: src/Plugins/Primitives/Plane/Makefile
Comment (Njord201) [inline src/Plugins/Primitives/Plane/Makefile:1]: Why edit the Makefile ?

**Review by Njord201 — CHANGES_REQUESTED** [review]

File: include/Parser/Factory.hpp (hunk 1 above)
Comment (AxelF44) [reply include/Parser/Factory.hpp:20]: Like I include sphere it's for dynamic cast

File: src/Plugins/Primitives/Plane/Makefile
Comment (AxelF44) [reply src/Plugins/Primitives/Plane/Makefile:1]: To compile the project, you can't create a dynamic library who use the Ray and Vector3d class whitout include them in compilation

**Review by mathieurobert1 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Pull request merged**